backend/cognito/template.yaml

```diff
@@ -4,7 +4,85 @@ Description: >-
 Transform:
   - AWS::Serverless-2016-10-31
 
+Parameters:
+  DomainOrEmailIdentityToVerify:
+    Type: String
+    Description: Domain or email that will be created as verified identity.
+  SignInHostedZone:
+    Type: String
+    Description: Deployment specific SignInHostedZone
+
 Resources:
+  DomainIdentityForSES:
+    Type: AWS::SES::EmailIdentity
+    Properties:
+      EmailIdentity: !Ref DomainOrEmailIdentityToVerify
+
+  DevelopmentEmailIdentityRoute53Records:
+    DependsOn:
+      - DomainIdentityForSES
+    Type: AWS::Route53::RecordSetGroup
+    Properties:
+      Comment: "Records created to verify domain ownership with SES"
+      HostedZoneId: !ImportValue
+       'Fn::Sub': '${SignInHostedZone}'
+      RecordSets:
+        - Name: !GetAtt DomainIdentityForSES.DkimDNSTokenName1
+          Type: CNAME
+          TTL: 86400
+          ResourceRecords:
+            - !GetAtt DomainIdentityForSES.DkimDNSTokenValue1
+
+        - Name: !GetAtt DomainIdentityForSES.DkimDNSTokenName2
+          Type: CNAME
+          TTL: 86400
+          ResourceRecords:
+            - !GetAtt DomainIdentityForSES.DkimDNSTokenValue2
+
+        - Name: !GetAtt DomainIdentityForSES.DkimDNSTokenName3
+          Type: CNAME
+          TTL: 86400
+          ResourceRecords:
+            - !GetAtt DomainIdentityForSES.DkimDNSTokenValue3
+
+  confirmForgotPasswordCustomMessage:
+    # checkov:skip=CKV_AWS_115: "Ensure that AWS Lambda function is configured for function-level concurrent execution limit"
+    # checkov:skip=CKV_AWS_117: "Ensure that AWS Lambda function is configured inside a VPC"
+    # checkov:skip=CKV_AWS_116: "Ensure that AWS Lambda function is configured for a Dead Letter Queue(DLQ)"
+    # checkov:skip=CKV_AWS_173: "Check encryption settings for Lambda environmental variable"
+    Type: AWS::Serverless::Function
+    Properties:
+      Policies:
+        - Version: '2012-10-17'
+          Statement:
+            - Effect: Allow
+              Action:
+                - 'ses:SendEmail'
+                - 'ses:SendRawEmail'
+              Resource: '*'
```

Comment:
Can we generate the ARN here for the EmailIdentity we've just created instead of a wildcard?

The example format of the ARN is: `arn:aws:ses:us-east-1:888888888888:identity/example.com` so it would need something like:

`!Sub arn:aws:ses:${AWS::Region}:${AWS::AccountId}:identity/${DomainOrEmailIdentityToVerify}` to work.

Where do we get the from address from?  I can't see where we actually pass it in.  And with that in mind, can we look at [restricting the from address](https://docs.aws.amazon.com/ses/latest/dg/sending-authorization-policy-examples.html#sending-authorization-policy-example-from) too while we're at it?

Reply:
It is all taken care @JohnWattsGDS. ses:SendEmail was added in two places in the template. Second one was unnecessary. I have removed it. So there is only one place where we need this !Sub arn:aws:ses:${AWS::Region}:${AWS::AccountId}:identity/${DomainOrEmailIdentityToVerify instead of '*'. Which I have tested and working okay after submitting the prod account access for SES.